cancer_ai/rewarder_test.py

```diff
@@ -0,0 +1,301 @@
+        expected_share_leader_1 += remaining_share_leader_3 / 2
+        expected_share_leader_2 += remaining_share_leader_3 / 2
+
+        # All should have the same shares
+        self.assertAlmostEqual(rewarder.scores["leader-1"], expected_share_leader_1, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-2"], expected_share_leader_2, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-3"], expected_share_leader_3, places=2)
+
+    def test_three_competitions_three_leaders_all_maximum_reductions(self):
+        """Test case 7: All competitors have maximum reduced shares (90%)"""
+        competitions_leaders = {
+            "competition-1": CompetitionLeader(hotkey="leader-1", leader_since=datetime.now() - timedelta(days=91)),
+            "competition-2": CompetitionLeader(hotkey="leader-2", leader_since=datetime.now() - timedelta(days=91)),
+            "competition-3": CompetitionLeader(hotkey="leader-3", leader_since=datetime.now() - timedelta(days=91))
+        }
+        scores = {"leader-1": 0, "leader-2": 0, "leader-3": 0}
+        rewarder_config = RewarderConfig(competition_leader_mapping=competitions_leaders, scores=scores)
+        rewarder = Rewarder(config=rewarder_config)
+
+        rewarder.update_scores()
+
+        base_share = 1 / 3
+        reduction_factor = 0.1  # 90% reduction for all
+
+        expected_share_leader_1 = base_share * reduction_factor
+        expected_share_leader_2 = base_share * reduction_factor
+        expected_share_leader_3 = base_share * reduction_factor
+
+        # Calculate distributed reduction
+        remaining_share_leader_1 = base_share - expected_share_leader_1
+        remaining_share_leader_2 = base_share - expected_share_leader_2
+        remaining_share_leader_3 = base_share - expected_share_leader_3
+
+        # Leaders 2 and 3 gets their base share plus the distributed reduction from Leader 1
+        expected_share_leader_2 += remaining_share_leader_1 / 2
+        expected_share_leader_3 += remaining_share_leader_1 / 2
+
+        # Leaders 1 and 3 gets their base share plus the distributed reduction from Leader 2
+        expected_share_leader_1 += remaining_share_leader_2 / 2
+        expected_share_leader_3 += remaining_share_leader_2 / 2
+
+        # Leaders 1 and 2 gets their base share plus the distributed reduction from Leader 3
+        expected_share_leader_1 += remaining_share_leader_3 / 2
+        expected_share_leader_2 += remaining_share_leader_3 / 2
+
+        # All should have the same shares
+        self.assertAlmostEqual(rewarder.scores["leader-1"], expected_share_leader_1, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-2"], expected_share_leader_2, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-3"], expected_share_leader_3, places=2)
+
+    def test_three_competitions_two_competitors(self):
+        """Test case 8: 3 competitions but only 2 competitors"""
+        competitions_leaders = {
```

Comment:
it'd be nice to name variable 
`leader_1_since = datetime.now() - timedelta(days=21) # 10% reduction` for clarity, otherwise you need to dig into code to see what reduction and why